Switch to patches and update

pvuorela opened the pull request:
Over a decade worth of digressed work in this repository compared to upstream. And one of the few remaining cases of the git subtree setup.  Started entangling the changes as rebased patches in hope that we could move more work to upstream. In past years there's been maybe more activity on that side. Alongside generic maintenance e.g. on the build system.

There's been also small divergence on the data format. Some almost identical elements added but order specified slightly different. At this point getting now rid of such a difference on 'authentication', with the three usages of it moved slightly forward in the xml tree accordingly. Hoping we could avoid all this kind of differences.

On the rebase work, biggest painpoint has been the commit at the start "Added MMS settings" by Slava which shared a lot of content with the upstream commit "Add MMS information (bgo #673744)" also by Slava. For the differences I interpreted it so that our commit being a few months newer was a revised version.

Quite a bit of commit and subsequent fix for it have been squashed. Some defunct operators dropped altogether from the delta. More details on separate commit which are now updating forward an upstream release at a time.

The plan onwards was to continue rebasing until the latest upstream, feed some local changes to upstream and eventually get the delta a lot smaller. For later consideration do we want to keep patches here or should we then switch to pulseaudio/bluez style git setup with occasionally rebased real commits.

Disclaimer: testing so far is that the xml validates and the rpm builds. Actually not sure how much I can even test.

At this point this takes to end of 2020. Thought I could already make a PR in case we end up discussing the generic details and the version here is already well beyond halfway.

pvuorela committed:
[mobile-broadband-provider-info] Update checkout a bit and regenerate delta as patches. JB#61583

Trying to entangle roughly a decade of digressed work on the repository.
Extracted the local changes as patches and nudged the upstream checkout
a few commit forward to include Slava's
"Add MMS information (bgo #673744)"
which roughly matched a big commit from Slava here
"Added MMS settings".

Working with the patches has its pros and cons but at least now the
delta is cleaner and we can continue moving towards the upstream which
has had bunch of work done.

Later can consider should we use a sailfish pulseaudio/bluez style
setup where local change are real commit but allowing to properly
rebase them against upstream.

While doing this, I cleaded up some changes going back and forth:
- 'Shorten excessively long access point names' -> defunct saunalahti,
removed.
- defunct aina parts removed
- defunct tele finland removed
- movistar argentina using upstream info
- 'Added comment for 3 HK' -> squash to 'Use LTE APN for 3HK'
- 'Fixed MMS proxy address for Cricket Wireless' -> squash 'Settings
for Cricket Wireless'
- 'Corrected apn settings for Jio 4G (India)' -> 'Added apn settings
for Jio 4G'
- 'Updated MMS settings for Free Mobile (France)' -> 'Replaced
mms.free.fr with its IP address'
- 'Fixed OneCall (Norway) operator name' -> Added information abouot
One Call Norway.
- 'Updated O2 UK MMS settings' -> 'Add O2 UK MMS settings'
- 'Updated settings for Bolivia' -> 'Added settings for Bolivia.'
- 'Fixed xml validation errors' -> previous 'Update
serviceproviders.xml' about azeri providers
- 'Fixed xml validation errors' -> previous 'United Arab Emirates
providers update'

And quite a bit squashed to 'Added MMS settings':
- 'Remove spaces from URLs'
- 'MMS settings for Etisalat (UAE)'
- 'Fixed MMS proxy port number for Elisa/Estonia'
- 'MMS settings for T-Mobile USA'
- 'Add T-Mobile USA MMS settings'
- 'Cleaned up Tele2 EE settings'
- 'Fixed MMS settings for Orange Israel'
- 'Updated settings for China Mobile HK'
- 'MMS proxy for Three (3) Sweden'
- 'Updated Telenor DK MMS settings'
- 'LMT Latvia MMS access point update'
- 'Vodafone Romania APNs updated'
- 'Updated settings for WIND Greece'
- 'Removed Telenor MMS settings from 3 Sweden block'
- 'Bibob DK MMS settings no longer work, took them out'
- 'Aster PL is no more'
- 'Fix MMS proxy IP for Telenor Sweden'
- 'Got rid of redundant Beeline'
- 'Update serviceproviders.xml' / 'Correcting French provider
"SFR" MMS setting'

pvuorela committed:
Remove local copy of the data

pvuorela committed:
Update to 20151214

- The authentication element dtd addition now in upstream though
in slightly different order. Dropped the local patch and made the
3 instances follow upstream way.
- Giffgaff config was doing some adjustments in MMS settings patch
and later fixing in separate one. Squashed latter to former.

pvuorela committed:
Update to 20170310

- Squashed 'Removed unused elements' -> 'Update T-Mobile(Telekom)'.
Was a cleanup commit to previous.
- Squashed some finnish adjustments to main MMS settings as there
were changes back and forth on those and upstream.

pvuorela committed:
Update to 20190116

pvuorela committed:
Update to 20190618

pvuorela committed:
Update to 20201225

- squashed fixing DTD to main mms. Was in fact creating a problem
and then fixing it.
- squashed 'Removed non-functional MMS AP for Vodafone Portugal'
to main mms which was introducing it.
- dropped 'MMS settings for Drei' which were actually just removing
some three.co.uk apns from it. Those were introduced in upstream
MMS settings commit but not included in local MMS settings commit.
Not sure if we that delta. If we'd do, that shouldn't be in this
patch anyway.

pvuorela commented:
The removed dns commit, do we really need/want that? Commit message says "It's not being used by ofono and is hopelessly out of date." -- even if it's out of date, it could be simpler just skipping that delta?

pvuorela committed:
Update to 20210805

Squash 'Update T-Mobile' -> 'LTE compatible configuration for Telekom DE'.
Upstream got most of the changes.

pvuorela committed:
Update to 20220315

- 'Added MMS settings for Hot Mobile (Israel)' squashed to
main MMS commit. upstream got similar changes and only some minor
naming difference was left.
- Avoiding deletion of one orange france mms apn to which
upstream added mmsc/mmsproxy
- Removed some main mms changes for removed canadian windmobile,
 and chilean operators (mms no longer offered it seems)
- Cleaned up patches e.g. not to add mnc 150 to AT&T and then removing it
in Cricket Wireless patch.

pvuorela committed:
Update to 20221107

- orange.co.il mms now upstream
- US AT&T apns updated upstream, using those now
- british t-mobile and orange had merger as EE upstream.
removed related changes here.

pvuorela committed:
Update to 20230416

- bunch of russian operators dropped
- <country> requires now <name>. Included in the additions now.

pvuorela committed:
Update to 20240407

- Quite a bunch of indian changes, old operators removed, new added.
Jio patch no longer needed.
- Switched to meson.

pvuorela committed:
[mobile-broadband-provider-info] Update to 20250613. JB#61583

Or actually current main branch, including one commit after the tag.

Here portuguese data had upstream changes so now
'Updated Vodafone PT settings' is more or less upstream, and
'Portugal cleanup' ended up just adding one name, thus just merged
to main MMS patch.

pvuorela commented:
Updated now to latest main. Phew.

The remaining diff is partly suspicious and the patches could be further cleaned up and combined, but hoping this is already for the better. The diff has been gradually getting smaller when getting towards upstream, e.g. there's been some changes now in upstream and there was some local changes for defunct operators now removed upstream.

monich commented:
be sure not to lose the `<protocol>` element and the `any` authentication method which don't exist upstream. those are actually being used.

pvuorela commented:
The protocol element is still there now on its own patch. Maybe something that could be suggested upstream? Btw, now content is "ip|ipv6|ipv4v6", was pondering should it be "ip" or rather "ipv4" for clarity?

The "any" method did actually get dropped in favor of just using the upstream declaration for the element. Did that as the xml file didn't,  and doesn't seem hasn't, have any usage for it. Does look like ofono side is parsing that  but is there a real need if the actual xml file doesn't use it?

Having said that, should be also simple enough to just have a small patch and keep it explicit that we have it handled elsewhere.

monich commented:
`ip|ipv6|ipv4v6` is what's historically been used as a parameter to `AT+CGDCONT`, in that sense it's quite logical. I chose not to introduce any changes based on aesthetical reasons, but rather follow the naming conventions that already existed in the modem world. at this point I'd say it is what it is, I don't see a good enough to make changes (especially such that are incompatible with the previous version)

pvuorela commented:
Sure, fair enough. Was just thinking if that's suggested upstream should it then be in current form or adjusted. I suppose the usage is anyway mostly the one file in our ofono so relatively simple to adjust.

pvuorela committed:
Add back auhentication method 'any'

Originally introduced in our version of the 'auhtentication' element,
but briefly dropped when switching to upstream declaration.
It's not used in the data but sailfish ofono does have handling,
so keeping the declaration here as well.

pvuorela commented:
Alright. Added a commit for bringing back the 'any' authentication method.
